### DISCUSSION TITLE
MPM for surface use

### DISCUSSION THREAD

**Discussion by WarrenHammerstrom** [pascallanger/DIY-Multiprotocol-TX-Module#653] [Ideas]
Hello!

New here, but I would like to start off with some questions about starting a project to do a mpm surface tx, tailored to more towards the popular protocols for cars, trucks, and cralwers.

I have no experience with coding, but I do have an understanding of electrical components and rf theory from my career as an aircraft mechanic and rc hobbiest. That said, my lack of experience in this feild will be made up for with enthusiasm and my desire to make a working prototype of this radio project.

My goal is to create a module that I can bring to a manufacturer ( I have been in limited talks with radiomaster and jumper) and form an agreement where we can get something of a product built.

With your interest and support, i hope we can sit down together with radiomaster and come up with a project plan.

Thank you for reading my proposal, and I hope this introduction peeks your interest and we can discuss possible ideas on where to go with it.

Take care, and thanks again!

**Comment by pascallanger** [comment]
Hi,
To reverse and add the protocols do not need a new module... The MPM already supports multiple surface protocols: Shenqi, DSMR, Kyosho/FHSS, Kyosho/Hype, Losi, Radiolink/Surface, Radiolink/DumboRC, V2x2 and Flysky.
What I really need is access to the cars/boats/motorcycles/... (at least the TX+RX) so I can reverse them. Usually people are sending me the hardware and I send it back once done (at their charges), purchase the model and send it to my attention, or make a substancial donnation (50% or more) so I can buy it.
Also note that I do already have good contact with RadioMaster and Jumper.
Pascal

**Reply by WarrenHammerstrom** [reply]
Thanks for the reply Pascal!

I have been talking with James at Radiomaster at length about this idea, but because I'm outside the company there isn't much he can do or say without getting himself in trouble. 
I'm more than happy to help, what would you need to reverse flysky AFHDS3, and Turbo Racing FHSS? Horizon hobby uses turbo racing gear in the minit 2.0 electronics, and I'm not certain, but a friend I chat with at crossover rc says the flysky 3 protocol uses a different chip compared to ahfds/afhds2.
If you would like, add me on facebook messenger, rhis way we can arrange itens or donations to get started. We can also discuss how to start the project with radiomaster as you have that working relationship in place. I know that your interest is only the mpm side of things, but the company and project together could come up with an idea for a working prototype transmitter.

Let me know what you think or need and i'll do my best to help.

**Reply by pascallanger** [reply]
AFHDS3 needs a SX12xx which the current multi module do not have but that shouldn't be a problem to add it since there is even a SPI port available (future expansion) for that on the PCBs. I have no idea what the protocol looks like and like with any protocols their might be some you can't figure out.
Turbo Racing is a big/huge challenge. I've already looked at their protocol. I'm not sure to be able to reverse it at all. To tell the truth I gave up... But I might have another look when I'll have time.

**Reply by WarrenHammerstrom** [reply]
So you have the stuff already for turbo racing, and looks like you've played with afhds3 too. Is there anything you need specifically that i can help you with?

**Reply by pascallanger** [reply]
Not entirely true. I've never seen an afhds3 TX or RX...

**Reply by WarrenHammerstrom** [reply]
I could send you my nb4 and fgr4s rx if you'd like.
I had to hack rhe radio, the 18650 board in the transmitter was faulty, had to do a bypass to get it powered by the powerbank. Not a huge deal, it works as it should.

Are you located in north america?

**Reply by pascallanger** [reply]
I'm located in France

**Reply by WarrenHammerstrom** [reply]
Find me on fb and message me, we'll arrange something

Warren Lee Hammerstrom.

I tried finding you, but there are a ton of pascal langers lol.

**Reply by WarrenHammerstrom** [reply]
Might actually be worth it to just have you buy it in france, shipping with the battery is astronomical.

Can you source one from a hobby store nearby and tell me what the bill is?

**Reply by pascallanger** [reply]
Which TX and RX model should I look for since I don't know their product line at all?

**Reply by WarrenHammerstrom** [reply]
Flysky noble nb4. Should come with a rx.
They have a new vwrsion called the pro, but im sure the protocol hasnt changed from version 1 of the tx.

https://www.flysky-cn.com/nb4

https://www.rcmodelz.co.uk/flysky-fsnb4-2-4ghz-transmitter-receiver.html

Maybe at the same time, we can try to find a turbo racing transmitter and rx, or the stock radio and esc from a losi mini-t 2.0. The stock gear is usually on fb marketplace or ebay for cheap because people swap it out for a dynomite syatem.

**Reply by WarrenHammerstrom** [reply]
https://www.ebay.ca/itm/TURBO-2-4GHz-91803G-91803G-VT-3CH-Radio-system-Transmitter-Controller-remote-con-/124803053473?mkcid=16&mkevt=1&_trksid=p2349624.m46890.l6249&mkrid=711-127632-2357-0

**Reply by WarrenHammerstrom** [reply]
I get the impression that radiomaster is cooking up a surface radio. Not that they said anything directly, but the way they were  saying they cant discuss any ongoing projects, you can read between the lines and guess that they are.
If you get these protocols added to the current mpm, can they be added later during a firmware update?
Or do you think this may be a new module altogther?

Im justt worried they are going to go ahead with whatever they are doing and possibly miss adding these protocols and possibly other ones that people may want.

I havent even looked into the orlandoo hunter stuff yet, that was something i was going to look into tonight.

**Reply by WarrenHammerstrom** [reply]
Apparently, orlandoo hunter might be using flysky, the model number they used for the fcc id is model No.: SI-4EE

**Reply by WarrenHammerstrom** [reply]
On second thought, maybe the orlandoo hunter stuff isnt flysky. Could you look into that fcc id for me?

Did you source somewhere to buy those radios? Im ready with etransfer when you have the total bill for me. After youre done with the transmitters you can either keep them,or sell them to put money into the project as a donation.

**Reply by WarrenHammerstrom** [reply]
Just thought I would check in and see how things are going. Haven't seen a reply in a month, hoping that means you're busy doing something hush hush :)

Let me know how things are going, still very interested in helping this idea along.

**Reply by pascallanger** [reply]
I think I know someone with an Orlandoo car. I'll ask him. Looking quickly on the web I haven't found the TX and RX to sell on BG or HK...

**Reply by WarrenHammerstrom** [reply]
How difficult would it be to get it from the UK?

https://www.rcmodelz.co.uk/flysky-fsnb4-2-4ghz-transmitter-receiver.html

**Reply by WarrenHammerstrom** [reply]
Whats the word on radiomaster, are they currently working on something?

**Reply by pascallanger** [reply]
I currently have familly issues and unless this is a simple protocol I haven't enough time to look into it. AFHDS3 is one of the protocols that would take a lot lot lot of time, so I'm not keen to start looking at it now. Other protocols using conventionnal RF chips would be easier and I can look into those but it will still take time before I deliver anything working in the current conditions.

**Reply by WarrenHammerstrom** [reply]
Sorry to hear that you're having difficulties at home. Family is a much bigger priority than designing toys.

**Reply by WarrenHammerstrom** [reply]
Hello!
I thought I would come back and check in and see how everyone is doing.
I was cuirous if you guys are back into the swing of things are perhaps we can get this idea rolling again!

**Reply by pascallanger** [reply]
I'm open adding surface protocols, but I need the transmitters and associated receivers (or models).

**Reply by WarrenHammerstrom** [reply]
I might be able to get you a few to start.
It would be easier if i could loop in my friend Ke Shan Hui who does alot of work at mxo-rc. He has disected alot of protocols that we can update the multi module with. I just don't know if he would want an agreement in place for lending his work. I'll have a talk with him

**Reply by bd4** [reply]
I am interested in this as well - I think there are two tasks here, one is to identify what R/C cars/trucks/crawlers are already supported by figuring out what protocols they use, and the other is adding new protocols that are not yet supported. I just noticed that the Axial SCX24 proto is already supported, I am going to try setting that up with my RadioMaster Boxer.
